compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2688,6 +2717,21 @@ private ParserRuleContext getNextRuleForKeywords(ParserRuleContext currentCtx, i
                     endContext(); // end let-clause-let-var-decl
                 }
                 return ParserRuleContext.EXPRESSION;
+            case ORDER_KEYWORD:
+                parentCtx = getParentContext();
+                if (parentCtx == ParserRuleContext.LET_CLAUSE_LET_VAR_DECL) {
+                    endContext(); // end let-clause-let-var-decl
+                }
+                return ParserRuleContext.BY_KEYWORD;
+            case BY_KEYWORD:
+                return ParserRuleContext.ORDER_KEY;
+            case ASCENDING_KEYWORD:
+            case DESCENDING_KEYWORD:
+                parentCtx = getParentContext();
+                if (parentCtx == ParserRuleContext.ORDER_CLAUSE_END) {
+                    endContext(); // end order-key
```

Comment:
I think you'll have to end the context which-ever path it takes, otherwise there will be nested contexts started without ending them